cmd/sync-tags/gomod.go

```diff
@@ -123,6 +107,24 @@ func updateGomodWithTaggedDependencies(tag string, depsRepo []string, semverTag
 			fmt.Printf("Warning: dependency %s not found in go.mod.\n", dep)
 		}
 	}
+
+	downloadCommand2 := exec.Command("go", "mod", "download")
+	downloadCommand2.Env = append(os.Environ(), "GO111MODULE=on", fmt.Sprintf("GOPRIVATE=%s", depPackages), "GOPROXY=https://proxy.golang.org")
+	downloadCommand2.Stdout = os.Stdout
+	downloadCommand2.Stderr = os.Stderr
+	if err := downloadCommand2.Run(); err != nil {
+		return changed, fmt.Errorf("error running go mod download: %v", err)
+	}
+
+	tidyCommand := exec.Command("go", "mod", "tidy")
```

Comment:
IIRC we decided to run `go mod tidy` after each update because we wanted to ensure that tags were correctly packaged and didn't contain errors.

If we run `go mod tidy` after all dependency updates, go will still check that all new tags are correctly packaged but I think we can't be sure if it removed any lines (not really sure where we might hit this case though)...

Having said that, we can't be sure even in the current case because subsequent `go mod tidy`s might still remove our updates, right? :thinking: 

Slightly tangential, but we might also want to remove `changed` and consider it as true always to help with https://github.com/kubernetes/publishing-bot/issues/187#issuecomment-564595877 and https://github.com/kubernetes/publishing-bot/issues/215. Thoughts?

https://github.com/kubernetes/publishing-bot/blob/ff2d8bb9188faeb5b4dd9369cae300e6c654c015/cmd/sync-tags/main.go#L330-L336